### PR TITLE
Fix proof matching.

### PR DESCRIPTION
**Pull request by mattcollier**
You'll let me know if the tests were not patched properly.

### CODE REVIEW

File: lib/ProofSet.js [hunk 1]
```diff
@@ -306,9 +306,10 @@ async function _verify({
   document, suites, proofSet, purpose,
   documentLoader, expansionMap, compactProof}) {
   // filter out matching proofs
-  const matches = proofSet.filter(proof => purpose ?
+  const result = await Promise.all(proofSet.map(proof => purpose ?
```
Comment (dlongley) [inline lib/ProofSet.js]: `purpose` is always defined so we can remove the conditional and just always call `purpose.match`. I think this was some noise left over from legacy code.

File: lib/ProofSet.js [hunk 2]
```diff
@@ -306,9 +306,10 @@ async function _verify({
...
     purpose.match(proof, {document, documentLoader, expansionMap}) :
-    true);
+    true));
+  const matches = proofSet.filter((value, index) => result[index]);
```
Comment (dlongley) [inline lib/ProofSet.js]: Why are we using `index`? Does the order change somehow?

Comment (mattcollier) [reply lib/ProofSet.js]: we are filtering `proofSet` based on `result`.  Is there a better way?

File: lib/ProofSet.js (hunk 1 above)
Comment (mattcollier) [reply lib/ProofSet.js]: Done.

File: lib/ProofSet.js (hunk 2 above)
Comment (dlongley) [reply lib/ProofSet.js]: Hmm, I guess it's fine. I was thinking we would map to the `proof` or `null` based on the result of awaiting `match` during the `map()` call and then filter out based on `null` or not (like below). But I suppose this gets the job done fine.

**Pull request merged**